src/test/java/com/waverleysoftware/AccessCodeTests.java

```diff
@@ -1,32 +1,31 @@
 package com.waverleysoftware;
 
-import org.testng.annotations.AfterClass;
-import org.testng.annotations.BeforeClass;
+import com.waverleysoftware.model.Panel;
 import org.testng.annotations.Test;
 
-public class AccessCodeTests extends BaseTest {
-    @BeforeClass
-    public void setUp() {
-        System.out.println("Setting up on child level");
-    }
-
-    @Test(priority = 1)
-    public void accessCodeShouldBeCreated() {
-        System.out.println("Access code created");
-    }
+import java.util.logging.Level;
 
-    @Test(priority = 2)
-    public void accessCodeShouldBeModified() {
-        System.out.println("Access code modified");
-    }
+/**
+ * A simple model class for access code testing.
+ */
+public class AccessCodeTests extends BaseTest {
 
-    @Test(priority = 3)
-    public void accessCodeShouldBeDeleted() {
-        System.out.println("Access code deleted");
+    @Test(dataProvider = "getPanelData", dataProviderClass = DataSuppliers.class)
+    public void accessCodeShouldBeAdded(final Panel panel) {
+        final String conditionForType = "home";
+        if (conditionForType.equals(panel.getType())
+                && panel.hasTouchScreen()
+                && LOG.isLoggable(Level.INFO)) {
```

Comment:
with `@Slf4j` you don't need to check of corresponding logging type is available

Reply:
I did it in the  **5-lombok-assert-annotation** branch. Please, look at the changes.